### ISSUE TITLE
Enchantment Level Predictions, for simpler verification of XP seed cracking. (Feature request)

### ISSUE THREAD

**Issue by HaphLife** [Earthcomputer/EnchantmentCracker#244]
There are lots of ways that the user can mess up using this tool, and here's a few suggestions of changes that would allow the user to fix those errors, or never make them in the first place:

A: Enchantment level requirement predictions: Once you complete cracking an XP seed, the tool should predict all of the remaining enchantment level requirements, so that if the user cares to they can check if those are correct and become more certain that the XP seed is correctly cracked. Additionally, while doing dummy and wanted enchantments, it should continue to predict the enchantment level requirements, so that the user can tell they messed up before enchanting a wanted item (which currently wastes levels, lapis, and even the item pre-grindstone).

**Comment by HaphLife** [comment]
How do I add labels?

**Comment by Earthcomputer** [comment]
You can't, I have to do it. Please can you split your suggestions into separate issues? It makes it easier for me to track them individually.

**Comment by HaphLife** [comment]
Okay.
How granular do you want it. Is A-E good enough or should I split them up more?

**Comment by Earthcomputer** [comment]
A-E is probably good enough

**Comment by HaphLife** [comment]
Ok. On it.

**Comment by HaphLife** [comment]
Could you re-open this, I editted it down to 1 suggestion. I could make a new version of this, but then the letters would be out of order. I guess that would let me fix the name.

**Comment by HaphLife** [comment]
Ooh, I can edit the name. yay!

**Comment by HaphLife** [comment]
Thank you :)

**Comment by Earthcomputer** [comment]
The levels on the right will be the same as the input after you crack it, since there is no dummy enchantment that inherently comes after the cracking process. However it would be a good idea to display the XP levels before each actual enchantment, it would add a good way to verify before actually enchanting the item, which is particularly important pre-grindstone.

**Comment by HaphLife** [comment]
I was meaning for the other shelf options. If I input shelves 15-10 to crack xp seed, and it then predicts shelves 9-0, I can check some of the remaining ones to verify it was the correct xp seed, and I didn’t get unlucky and make an incorrect input that was actually possible with a different seed.

**Comment by Earthcomputer** [comment]
If you input incorrect data, it is much more likely to fail with 0 seeds remaining rather than find an incorrect seed. In this case, you can simply reset the cracker and crack the XP seed again (without losing progress). Doing better than this is covered by #245.

**Comment by HaphLife** [comment]
Fair enough. As I mentioned in #245 it would be good to put these predictions in the same spot as over there, so that it doesn’t take up any extra UI space. The difference between the main point of this and the main point of #245 is that when you use the list in #245 you know you did something wrong, the xp seed cracking failed, but this is to reduce risk of false positives. I guess you’ll find out there was a false positive when you try to crack the player seed, but this lets you know which xp seed was wrong, and if it’s the first seed, lets you know while you can fix it.

**Comment by HaphLife** [comment]
I don’t know how common the xp-seed false positive is, but it certainly is an edge-case.

**Comment by Earthcomputer** [comment]
Failing to crack the XP seed is usually due to doing banned actions in between the two XP seeds, playing on a paper server, or doing the steps in the wrong order. It's usually not a problem with either XP seed themselves.
